processing/src/main/java/io/druid/query/metadata/metadata/SegmentMetadataQuery.java

```diff
@@ -121,11 +117,17 @@ public SegmentMetadataQuery(
     if (querySegmentSpec == null) {
       this.usingDefaultInterval = true;
     } else {
-      this.usingDefaultInterval = useDefaultInterval == null ? false : useDefaultInterval;
+      if (querySegmentSpec.getIntervals().size() == 1 && querySegmentSpec.getIntervals()
```

Comment:
Please address this. Also test that if the default interval is *explicitly* specified via query builder (rather than leaving it as `null`, that will be replaced with the default interval in the `SegmentMetadataQuery`'s constructor), than isDefaultInterval is correctly set to `true`.